packages/console/src/Components/InputTime/InputTime.tsx

```diff
@@ -41,25 +44,33 @@ function InputTime(props: IInputTime) {
    * Triggers change to the input outside.
    * This function will format the date using the time that we selected and the time format as well.
    */
-  function triggerChange() {
-    if (usesAmPmFormat) {
-      props.onChange(`${hour || "00"}:${minute || "00"} ${format}`);
-    } else {
-      props.onChange(`${hour || "00"}:${minute || "00"}`);
-    }
-  }
+  const triggerChange = useCallback(
+    (params: {
+      hour: string;
+      minute: string;
+      format: string;
+    }) => {
+      if (usesAmPmFormat) {
+        props.onChange(
+          `${params.hour || "00"}:${params.minute || "00"} ${params.format}`,
+        );
+      } else {
+        props.onChange(`${params.hour || "00"}:${params.minute || "00"}`);
+      }
+    },
+    [usesAmPmFormat, props.onChange],
+  );
 
   /**
    * Uses to trigger a change to the input outside if the hour, minute or time format has changed.
    * We ignore the first one because it's not needed.
    */
   useEffect(() => {
     if (!firstRender.current) {
-      triggerChange();
+      triggerChange({ hour, minute, format });
```

Comment:
I've fixed this in other places as well, making:

1. `useEffect` to track the dependencies
2. Function doesn't use state in dependency array, receives as params